distributed/tests/test_client.py

```diff
@@ -7345,6 +7336,109 @@ async def test_dump_cluster_state_json(c, s, a, b, tmp_path, local):
         await c.dump_cluster_state(filename, format="json")
 
 
+def blocked_inc(x, event):
+    event.wait()
+    return x + 1
+
+
+@pytest.mark.parametrize("local", [True, False])
+@pytest.mark.parametrize("_format", ["msgpack", "yaml"])
+@gen_cluster(client=True)
+async def test_inspect_cluster_dump(c, s, a, b, tmp_path, _format, local):
+    filename = tmp_path / "foo"
+    if not local:
+        pytest.importorskip("fsspec")
+        # Make it look like an fsspec path
+        filename = f"file://{filename}"
+
+    futs = c.map(inc, range(2))
+    fut_keys = {f.key for f in futs}
+    await c.gather(futs)
+
+    event = distributed.Event()
+    blocked_fut = c.submit(blocked_inc, 1, event)
+
+    await c.dump_cluster_state(filename, format=_format)
+
+    scheduler_tasks = list(s.tasks.values())
+    worker_tasks = [t for w in (a, b) for t in w.tasks.values()]
+
+    event.set()
+
+    suffix = ".gz" if _format == "msgpack" else ""
+    dump = DumpArtefact(f"{filename}.{_format}{suffix}")
+
+    smem_keys = {t["key"] for t in dump.scheduler_tasks("memory")}
+    wmem_keys = {t["key"] for t in dump.worker_tasks("memory")}
+
+    assert smem_keys == fut_keys
+    assert smem_keys == {t.key for t in scheduler_tasks if t.state == "memory"}
+    assert wmem_keys == fut_keys
+    assert wmem_keys == {t.key for t in worker_tasks if t.state == "memory"}
+
+    task_key = next(iter(fut_keys))
+
+    expected = [
+        (task_key, "released", "waiting", {task_key: "processing"}),
+        (task_key, "waiting", "processing", {}),
+        (task_key, "processing", "memory", {}),
+    ]
+
+    story = dump.scheduler_story(task_key)
+    assert len(story) == len(expected)
+
+    for event, expected_event in zip(story, expected):
+        for e1, e2 in zip(event, expected_event):
+            assert e1 == e2
```

Comment:
Why not just `assert story == expected`?

Reply:
Scheduler stories also contain `stimulus_id`'s and timestamps which are non-deterministic for comparison purposes. Only the deterministic values are compared as the `zip(event, expected_event)` prunes out the `stimulus_id`'s and timestamps which occur at the end of the event tuples.